HarClient/RedirectPolicy.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+
+namespace JBlam.HarClient
+{
+    /// <summary>
+    /// Defines a policy for handling redirect messages between a <see cref="DelegatingHandler"/>
+    /// and its inner handler.
+    /// </summary>
+    public readonly struct RedirectPolicy : IEquatable<RedirectPolicy>
+    {
+        /// <summary>
+        /// The outer handler will attempt to suppress any automatic redirect behaviour in the
+        /// inner handler, but will follow the redirects itself automatically.
+        /// </summary>
+        public static readonly RedirectPolicy SuppressAndFollow = new RedirectPolicy(true, true);
+        /// <summary>
+        /// The outer handler will attempt to suppress any automatic redirect behaviour in the
+        /// inner handler, and will return any redirect messages directly instead of following
+        /// them.
+        /// </summary>
+        public static readonly RedirectPolicy SuppressOnly = new RedirectPolicy(false, true);
+        /// <summary>
+        /// The outer handler will allow the inner handler to redirect automatically if it chooses;
+        /// it will return any redirect messages if they are observed.
+        /// </summary>
+        public static readonly RedirectPolicy DoNotSuppress = new RedirectPolicy(false, false);
+
+        public RedirectPolicy(bool followRedirects, bool suppressInnerAutoRedirect)
+        {
+            FollowRedirects = followRedirects;
+            SuppressInnerAutoRedirect = suppressInnerAutoRedirect;
+        }
+        /// <summary>
+        /// Gets the observable auto redirect policy. If true, the handler will follow redirects;
+        /// otherwise, it will return any redirect response messages directly.
+        /// </summary>
+        public bool FollowRedirects { get; }
+        /// <summary>
+        /// Gets the inner-handler auto redirect policy. If true, the inner handler's auto redirect
+        /// behaviour will be suppressed (where possible). Otherwise, the inner handler will behave
+        /// as it was originally configured.
+        /// </summary>
+        public bool SuppressInnerAutoRedirect { get; }
```

Comment:
Consider if we want this to be "requested inner behaviour" `bool?`

Reply:
Bearing in mind that "auto redirect" is a _feature_, it makes a certain amount of sense that we'd be _suppressing_ (disabling) _the feature_ here.

#WONTFIX